src/wallet/rpcdump.cpp

```diff
@@ -330,6 +331,26 @@ UniValue importwallet(const JSONRPCRequest& request)
         boost::split(vstr, line, boost::is_any_of(" "));
         if (vstr.size() < 2)
             continue;
+
+        // Sapling keys
+        // Let's see if the address is a valid PIVX spending key
+        if (pwalletMain->HasSaplingSPKM()) {
+            libzcash::SpendingKey spendingkey = KeyIO::DecodeSpendingKey(vstr[0]);
+            int64_t nTime = DecodeDumpTime(vstr[1]);
+            if (IsValidSpendingKey(spendingkey)) {
+                libzcash::SaplingExtendedSpendingKey saplingSpendingKey = *boost::get<libzcash::SaplingExtendedSpendingKey>(&spendingkey);
+                auto addResult = pwalletMain->GetSaplingScriptPubKeyMan()->AddSpendingKeyToWallet(
+                        Params().GetConsensus(), saplingSpendingKey, nTime);
+                if (addResult == KeyAlreadyExists){
+                    LogPrint(BCLog::SAPLING, "Skipping import of shielded addr (key already present)\n");
+                } else if (addResult == KeyNotAdded) {
+                    // Something went wrong
+                    fGood = false;
+                }
+                continue;
```

Comment:
it's needed to continue looping, at this point, the sapling spending key was successfully decoded. Without the continue, instead of moving to the next line directly, it will try to decode the string as a base58 privkey, fail and move to the next line.

Reply:
Yeah,  right, we are still inside `IsValidSpendingKey(spendingkey)`, all good.